### PR TITLE
Reformat files with scalafmt

### PR DESCRIPTION
**Pull request by clee704**
### What is the context for this pull request?
 - **Tracking Issue**: N/A
 - **Parent Issue**: N/A
 - **Dependencies**: N/A

### What changes were proposed in this pull request?
Reformat files with `scalafmt`, as in the following command:

```bash
$ scalafmt -c dev/.scalafmt.conf src
```

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Unit test

### CODE REVIEW

**Comment by paryoja** [comment]
> $ scalafmt -c dev/.scalafmt.conf src

What if we add this command to pre-commit hook?
Every committers should set their environment to use pre-commit hook so it cannot be mandatory though.

**Comment by clee704** [comment]
I think we can add a guideline in README or somewhere regarding scalafmt, and a git hook is a good one. There can be other ways, though, like running scalafmt automatically in your IDE or vim (using coc-metals).

**Comment by clee704** [comment]
Seems we can use `sbt-scalafmt` to check the formatting. What do you think about adding the check to the build or CI to enforce the formatting? @imback82 @sezruby

**Comment by paryoja** [comment]
> I think we can add a guideline in README or somewhere regarding scalafmt

I mentioned to use 'reformat on save' in the formatting guild committed in #471 but it wasn't obligation so I wonder how many committers read it carefully and set it :( 

sbt-scalafmt could be good. It is the project mentioned in the scalafmt guide. https://scalameta.org/scalafmt/docs/installation.html

File: src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala [hunk 1]
```diff
@@ -224,12 +224,10 @@ object Directory {
    * @param files List of leaf files.
    * @param fileIdTracker FileIdTracker to keep mapping of file properties to assigned file ids.
    *                      Note: If a new leaf file is discovered, the input fileIdTracker gets
-    *                     updated by adding it to the files it is tracking.
+   *                     updated by adding it to the files it is tracking.
```
Comment (sezruby) [inline src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala:227]:
```suggestion
   *                      updated by adding it to the files it is tracking.
```

**Review by sezruby — COMMENTED** [review]
Thanks for doing this 👍 Other than the comment, LGTM

File: src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala (hunk 1 above)
Comment (clee704) [reply src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala]: Done

**Review by sezruby — APPROVED** [review]

**Comment by clee704** [comment]
Opened #476 for enforcing/checking scalafmt during build/CI

**Pull request merged**